includes/admin/class-wp-job-manager-permalink-settings.php

```diff
@@ -135,26 +135,32 @@ public function settings_save() {
 			return;
 		}
 
-		if ( isset( $_POST['permalink_structure'] ) ) {
-			if ( function_exists( 'switch_to_locale' ) ) {
-				switch_to_locale( get_locale() );
-			}
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- WP core handles nonce check for settings save.
+		if ( ! isset( $_POST['permalink_structure'] ) ) {
+			// We must not be saving permalinks.
+			return;
+		}
 
-			$permalink_settings = WP_Job_Manager_Post_Types::get_raw_permalink_settings();
+		if ( function_exists( 'switch_to_locale' ) ) {
+			switch_to_locale( get_locale() );
+		}
 
-			$permalink_settings['job_base']      = isset( $_POST['wpjm_job_base_slug'] ) ? sanitize_title_with_dashes( wp_unslash( $_POST['wpjm_job_base_slug'] ) ) : '';
-			$permalink_settings['category_base'] = isset( $_POST['wpjm_job_category_slug'] ) ? sanitize_title_with_dashes( wp_unslash( $_POST['wpjm_job_category_slug'] ) ) : '';
-			$permalink_settings['type_base']     = isset( $_POST['wpjm_job_type_slug'] ) ? sanitize_title_with_dashes( wp_unslash( $_POST['wpjm_job_type_slug'] ) ) : '';
+		$permalink_settings = WP_Job_Manager_Post_Types::get_raw_permalink_settings();
 
-			if ( isset( $_POST['wpjm_job_listings_archive_slug'] ) ) {
-				$permalink_settings['jobs_archive'] = sanitize_title_with_dashes( wp_unslash( $_POST['wpjm_job_listings_archive_slug'] ) );
-			}
+		// phpcs:disable WordPress.Security.NonceVerification.Missing -- WP core handles nonce check for settings save.
+		$permalink_settings['job_base']      = isset( $_POST['wpjm_job_base_slug'] ) ? sanitize_title_with_dashes( wp_unslash( $_POST['wpjm_job_base_slug'] ) ) : '';
```

Comment:
Kinda similarly, these three (and maybe more in other places) could be refactored to a method

Reply:
Ditto for previous comment.